log/klogv2/klogv2.go

```diff
@@ -96,3 +112,22 @@ func (l *adapter) WithFields(fields log.Fields) log.Logger {
 func (l *adapter) WithError(err error) log.Logger {
 	return l.WithField("err", err)
 }
+
+// processFields prepares a string to be appended to every log call.
+// This is called once to avoid costly log operations.
+func processFields(fields map[string]interface{}) string {
+	// Order fields lexically for easier reading.
+	keys := make([]string, 0, len(fields))
+	for k := range fields {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	processedFields := make([]string, 0, len(fields))
+	for _, k := range keys {
+		processedFields = append(processedFields, fmt.Sprintf("%s=%+v", k, fields[k]))
```

Comment:
any reason to do this in two passes?
and not:

```go
for key := range fields {
  processedFields = append(processedFields, fmt.Sprintf("%s=%+v", key, fields[key]))
}

sort.Strings(processedFields)
```

Reply:
Nice catch!

Reply:
Done.